components/psst/browser/core/psst_component_installer.cc

```diff
@@ -0,0 +1,169 @@
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/functional/bind.h"
+#include "base/functional/callback.h"
+#include "base/functional/callback_forward.h"
+#include "base/no_destructor.h"
+#include "brave/components/psst/browser/core/psst_rule_service.h"
+#include "brave/components/psst/common/features.h"
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "crypto/sha2.h"
+
+namespace psst {
+
+namespace {
+
+// Directory structure of PSST component:
+// lhhcaamjbmbijmjbnnodjaknblkiagon/<component version>/
+//  |_ manifest.json
+//  |_ psst.json
+//  |_ scripts/
+//    |_ twitter/
+//        |_ test.js
+//        |_ policy.js
+//    |_ linkedin/
+//        |_ test.js
+//        |_ policy.js
+// See psst_rule.cc for the format of psst.json.
+
+constexpr size_t kHashSize = 32;
+const char kPsstComponentName[] =
+    "Brave Privacy Settings Selection for Sites Tool (PSST) Files";
+const char kPsstComponentId[] = "lhhcaamjbmbijmjbnnodjaknblkiagon";
+const char kPsstComponentBase64PublicKey[] =
+    "MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAphUFFHyK+"
+    "qUOXSw3OJXRQwKs79bt7zqnmkeFp/szXmmhj6/"
+    "i4fmNiXVaxFuVOryM9OiaVxBIGHjN1BWYCQdylgbmgVTqLWpJAy/AAKEH9/"
+    "Q68yWfQnN5sg1miNir+0I1SpCiT/Dx2N7s28WNnzD2e6/"
+    "7Umx+zRXkRtoPX0xAecgUeyOZcrpZXJ4CG8dTJInhv7Fly/U8V/KZhm6ydKlibwsh2CB588/"
+    "FlvQUzi5ZykXnPfzlsNLyyQ8fy6/+8hzSE5x4HTW5fy3TIRvmDi/"
+    "7HmW+evvuMIPl1gtVe4HKOZ7G8UaznjXBfspszHU1fqTiZWeCPb53uemo1a+rdnSHXwIDAQAB";
+
+}  // namespace
+
+class PsstComponentInstallerPolicy
+    : public component_updater::ComponentInstallerPolicy {
+ public:
+  PsstComponentInstallerPolicy();
+
+  PsstComponentInstallerPolicy(const PsstComponentInstallerPolicy&) = delete;
+  PsstComponentInstallerPolicy& operator=(const PsstComponentInstallerPolicy&) =
+      delete;
+
+  // component_updater::ComponentInstallerPolicy
+  bool SupportsGroupPolicyEnabledComponentUpdates() const override;
+  bool RequiresNetworkEncryption() const override;
+  update_client::CrxInstaller::Result OnCustomInstall(
+      const base::Value::Dict& manifest,
+      const base::FilePath& install_dir) override;
+  void OnCustomUninstall() override;
+  bool VerifyInstallation(const base::Value::Dict& manifest,
+                          const base::FilePath& install_dir) const override;
+  void ComponentReady(const base::Version& version,
+                      const base::FilePath& path,
+                      base::Value::Dict manifest) override;
+  base::FilePath GetRelativeInstallDir() const override;
+  void GetHash(std::vector<uint8_t>* hash) const override;
+  std::string GetName() const override;
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+
+  static base::NoDestructor<std::string> g_psst_component_id_;
+  static base::NoDestructor<std::string> g_psst_component_base64_public_key_;
```

Comment:
Is there any reason why these `NoDestructor` instances are required in this class? They don't seem to be referenced by anything else at the moment.